geps/gep-1709.md

```diff
@@ -37,8 +45,309 @@ which implementations can prove they satisfy and be recognized for.
+in the following format:
+
+```json
+apiVersion: v1alpha1
+kind: ConformanceReport
+implementation: acmeorg-acme
+reports:
+  layer4:
+    releases:
+    - tag: v0.7.0
+    - tag: v0.6.2
+    - tag: v0.6.1
+  layer7:
+    releases:
+    - tag: v0.7.0
+      extended:
+        unsupportedFeatures:
+        - HTTPResponseHeaderModification
+        - HTTPRouteQueryParamMatching
+    - tag: v0.6.2
+      extended:
+        unsupportedFeatures:
+        - HTTPResponseHeaderModification
+        - HTTPRouteQueryParamMatching
+        - HTTPRouteMethodMatching
+    - tag: v0.6.1
+      extended:
+        unsupportedFeatures:
+        - HTTPResponseHeaderModification
+        - HTTPRouteQueryParamMatching
+        - HTTPRouteMethodMatching
+maintainers:
+- @kubernetes-sigs/gateway-api-maintainers
+```
+
+> **NOTE**: In the above the `implementation` field is a combination of
+> `<organization>-<project>`. Organizations can be an open source organization,
+> an individual, a company, e.t.c.. Organizations can theoretically have more
+> than one `<project>` name and submit separate reports for each of them.
+
+> **NOTE**: The above demonstrates that the `acmeorg-acme` project was already
+> passing for versions prior to `v0.7.0`. This is how we will track history of
+> conformance for releases across implementations, but at some point might
+> require some archival process (which we probably don't have to worry about
+> for this iteration). You can also see that there's been progress to add the
+> `HTTPRouteMethodMatching` for the latest release.
+
+> **NOTE**: The `maintainers` field indicates the Github usernames or team
+> names of those who are responsible for maintaining this file, so they can be
+> easily contacted when needed (e.g. for relevant release announcements
+> regarding conformance, e.t.c.).
+
+TODO: need to add testing flags to emit the `ConformanceReport` here automatically
+
+The implementation can compile its `ConformanceReport` and upload it to the
+Gateway API project by creating a pull request. The file should be uploaded
+to the following sub-directory:
+
+```console
+conformance/results/<organization>-<project>.yaml
+```
+
+> **NOTE**: For this iteration we're focusing on getting these reports created
+> but at first they will probably be created by hand. In future iterations and
+> after having some time to experience the system, tooling to handle reporting
+> must be created.
+
+This will start the certification process which will result in the
+implementation receiving a badge they can use on their project pages and/or
+repositories which indicates their conformance levels and link to their
+conformance reports in the upstream Gateway API website.
+
+TODO: need to add some linting and validation for `ConformanceReports`
+
+TODO: need to add a merge tool to merge old reports into new ones
+
+> **NOTE**: No verification process (to prevent intentionally incorrect
```

Comment:
I think we could at least check that if an implementor sets a specific profile, all the tests belonging to such a profile are successfully executed. So, for example, if not running specific tests belonging to a profile through the `SkipTests` field, we might want to invalidate the certification as some required tests have not been run. WDYT?

Reply:
> I think we could at least check that if an implementor sets a specific profile, all the tests belonging to such a profile are successfully executed.

Yes, but more specifically all the _core_ tests. We are already making _core_ a minimum requirement:

https://github.com/kubernetes-sigs/gateway-api/pull/1754/files#diff-7764ccb47f68a80348b14a2c6df6c6cadd80db02de7817dbcb8039a611bf1328R169

> So, for example, if not running specific tests belonging to a profile through the SkipTests field, we might want to invalidate the certification as some required tests have not been run. WDYT?

Yes. There's a note specifically about this present here:

https://github.com/kubernetes-sigs/gateway-api/pull/1754/files#diff-7764ccb47f68a80348b14a2c6df6c6cadd80db02de7817dbcb8039a611bf1328R127

We're not 100% sure exactly how it will _look_ yet, but we are already trying to be clear that `SkipTests` usage will invalidate the `ConformanceReport` (or at least mark conformance as "in-progress") or something like that.

Reply:
Thanks for pointing out the notes mentioning this aspect, I somehow lost them :)